server/tserver/src/main/java/org/apache/accumulo/tserver/InMemoryMap.java

```diff
@@ -683,13 +683,6 @@ public void setInterruptFlag(AtomicBoolean flag) {
     private SourceSwitchingIterator ssi;
     private MemoryDataSource mds;
 
-    @Override
-    protected SortedKeyValueIterator<Key,Value> getSource() {
-      if (closed.get())
-        throw new IllegalStateException("Memory iterator is closed");
-      return super.getSource();
-    }
-
```

Comment:
I am going to experiment to see how much it affects performance and if there is a compromise where we only have the sanity check at the start of loops.